### PR TITLE
Fix prefix removal for folder paths with symlinks when copying a folder to S3

### PR DESCRIPTION
**Pull request by billowsofcode**
## Problem 

(May be related to open issue: https://github.com/soto-project/soto-s3-file-transfer/issues/20)

The folder argument passed to `copy(from folder: String, to s3Folder: S3Folder...` or `sync(from folder: String, to s3Folder: S3Folder,...` may be non-standardized and contain symlinks. In this case `targetFiles(files: [FileDescriptor], from srcFolder: String, to destFolder: S3Folder)` may not remove the folder prefix in all cases. It's because `fileEnumerator = FileManager.default.enumerator(` in `listFiles(in folder: String)` may return file paths that are standardized and with different use of symlinks (e.g. /private/var instead of /var).

Example illustrating problem:
```Swift

let temporaryFolder = try! FileManager.default.url(for: .itemReplacementDirectory,
                                                      in: .userDomainMask,
                                                      appropriateFor: URL(fileURLWithPath: NSString("~").expandingTildeInPath),
                                                      create: true)

let folder = temporaryFolder.appendingPathComponent("somefolder", isDirectory: true).appendingPathComponent("..")
print("folder.path: \(folder.path)")
print("folder resolved (fix): \(folder.standardizedFileURL.resolvingSymlinksInPath().path)")

try FileManager.default.createDirectory(at: folder, withIntermediateDirectories: true)
FileManager.default.createFile(atPath: folder.appendingPathComponent("file1.txt").path, contents: nil)

// assume folder is passed to `copy(from folder: String, to s3Folder: S3Folder...`.  That method will enumerate the files in the folder with paths like this:

guard let fileEnumerator = FileManager.default.enumerator(
    at: folder,
    includingPropertiesForKeys: [.isRegularFileKey],
    options: .skipsHiddenFiles
) else {
    fatalError("failedToEnumerateFolder")
}

while let file = fileEnumerator.nextObject() as? URL {
    if !file.hasDirectoryPath {
        print("file.path: \(file.path)")
        print("file resolved (fix): \(file.resolvingSymlinksInPath().path)")
    }
}
```

Output:
```
folder.path: /var/folders/9b/9ht20vfx44d6qy3hzg0_j0jc0000gp/T/TemporaryItems/NSIRD_com.apple.dt.Xcode.PlaygroundStub-macosx_ILT2Uc/somefolder/..
folder resolved (fix): /var/folders/9b/9ht20vfx44d6qy3hzg0_j0jc0000gp/T/TemporaryItems/NSIRD_com.apple.dt.Xcode.PlaygroundStub-macosx_ILT2Uc
file.path: /private/var/folders/9b/9ht20vfx44d6qy3hzg0_j0jc0000gp/T/TemporaryItems/NSIRD_com.apple.dt.Xcode.PlaygroundStub-macosx_ILT2Uc/file1.txt
file resolved (fix): /var/folders/9b/9ht20vfx44d6qy3hzg0_j0jc0000gp/T/TemporaryItems/NSIRD_com.apple.dt.Xcode.PlaygroundStub-macosx_ILT2Uc/file1.txt
```

## Solution

Run the same path standardization methods on both the user supplied folder argument and the enumerated file paths.  This way remove prefix will work in `targetFiles(`

### Side-Note

Working with file paths is difficult. Apple recommends to use URL instead of String for file paths. You should switch the API from using String to URL.  In your code only convert to a path string (URL.path) when a framework method requires it. https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/LowLevelFileMgmt/Articles/StandardDirectories.html

### CODE REVIEW

**Comment by adam-fowler** [comment]
Sorry haven't managed to get to this yet. I'll try tomorrow

File: Sources/SotoS3FileTransfer/S3FileTransferManager.swift [hunk 1]
```diff
@@ -265,7 +265,8 @@ public struct S3FileTransferManager {
         return listFiles(in: folder)
             .flatMap { files in
                 let taskQueue = TaskQueue<Void>(maxConcurrentTasks: configuration.maxConcurrentTasks, on: eventLoop)
-                let transfers = Self.targetFiles(files: files, from: folder, to: s3Folder)
+                let folderResolved = URL(fileURLWithPath: folder).standardizedFileURL.resolvingSymlinksInPath()
```
Comment (adam-fowler) [inline Sources/SotoS3FileTransfer/S3FileTransferManager.swift:268]: Any reason you don't put this inside of `targetFiles`

Comment (billowsofcode) [reply Sources/SotoS3FileTransfer/S3FileTransferManager.swift:268]: Yes I had this inside of `targetFiles` initially, but then I read the comments for all 3 `targetFiles` methods that say `Function assumes the files have srcFolder prefixed`. So I didn't want to change the assumption for one of them.  But I can move `let folderResolved = URL...` inside `targetFiles` if you wish.

Comment (adam-fowler) [reply Sources/SotoS3FileTransfer/S3FileTransferManager.swift:268]: Fair enough

**Review by adam-fowler — APPROVED** [review]

**Pull request merged**